### PR TITLE
Use jest for testing and eslint for linting

### PR DESCRIPTION
**Pull request by andreashouben**
Oops. I wanted to merge this in my repo before I create a pull request here. 😅 

Well, here it goes.

I took the liberty to upgrade the tech stack a bit.

I replaced mocha with jest. sinon and chai still work but can now be replaced by jests own mocking and matching functions.

Also I replaced jshint with eslint as it got better support for jest. I fixed a lot of linting errors on the way.

Because jest brings it's own reporting I could get rid of tee-pipefail.

Oh and I activated github actions because travis takes forever ;)

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        node-version: [8.x, 10.x, 12.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    - run: npm run build --if-present
```
Comment (basti1302) [inline .github/workflows/node.js.yml:28]: I don't think we need --if-present here.

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,5 @@
 .nyc_output/
 node_modules/
 npm-debug.log
+/.idea/
```
Comment (basti1302) [inline .gitignore:5]: Nitpick: Stuff like that should be in your ~/.gitignore, not in the .gitignore of each project. But whatever :-)

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 sudo: false
 language: node_js
 node_js:
-  - "8"
   - "10"
   - "12"
   - "13"
+  - "14"
```
Comment (basti1302) [inline .travis.yml:7]: should we remove .travis.yml completely when introducing github actions?

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
+/*
+ * For a detailed explanation regarding each configuration property, visit:
+ * https://jestjs.io/docs/en/configuration.html
+ */
+
+module.exports = {
+  // All imported modules in your tests should be mocked automatically
+  // automock: false,
```
Comment (basti1302) [inline jest.config.js:8]: Could you please remove all the commented out noise from the jest config template?

**Review by basti1302 — CHANGES_REQUESTED** [review]
This looks very good overall! A few small nitpicks. 

Thanks a ton for giving this whole repo the overhaul it badly needed. ❤️ ❤️

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
```
Comment (basti1302) [inline .github/workflows/node.js.yml:4]: Should the result from running this action be already visible in this PR. Do I need to enable it for the repo or something?

Comment (andreashouben) [reply .github/workflows/node.js.yml:4]: Nitpicking only shows that you have an eye for details. :) I will fix those issues asap.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (andreashouben) [reply .github/workflows/node.js.yml]: Removed.

File: jest.config.js (hunk 4 above)
Comment (andreashouben) [reply jest.config.js]: Done.

File: .github/workflows/node.js.yml (hunk 5 above)
Comment (andreashouben) [reply .github/workflows/node.js.yml:4]: If you look at my fork you'll see that it's already working. So it should, too when it's merged into your master.

https://github.com/andreashouben/repoman/actions

File: .travis.yml (hunk 3 above)
Comment (andreashouben) [reply .travis.yml:7]: I'd say yes.

File: .gitignore (hunk 2 above)
Comment (andreashouben) [reply .gitignore]: I did not know that git also looks into .gitignore files in the home directory. Neat. I removed this.

File: .travis.yml (hunk 3 above)
Comment (andreashouben) [reply .travis.yml:7]: Question is, should I do it in this PR? Or do you want to do it afterwards. Maybe run both pipelines once and if both are successful, remove the travis PL.

Comment (andreashouben) [reply .travis.yml:7]: Oh and also we should change the badge in the readme to point to the correct pipeline.

**Comment by basti1302** [comment]
Thanks a lot!

File: .travis.yml (hunk 3 above)
Comment (basti1302) [reply .travis.yml:7]: I've removed it and cherry picked the commit with the badge from your master into this.

**Pull request merged**